roles/varnish/templates/etc/varnish/varnish.vcl

```diff
@@ -176,89 +165,85 @@ sub vcl_init {
 {% endfor %}
 {% endfor %}
 
+    # Run builtin vcl_init
 }
 
 sub vcl_recv {
-
+    # Block any clients with an ip in the block acl
     if (client.ip ~ block) {
         return (synth(403, "Access denied"));
     }
 
+    # Remove multiple trailing /
+    if (req.url ~ "/{2,}$") {
+        return (synth(301, bereq.http.host + regsub(bereq.url, "/{2,}$", "/")));
+    }
+
+    # Check purge ACL based on `req.http.x-forwarded-for` instead of `client.ip`
+    # because `client.ip` has the IP address of haproxy.
+    # `req.http-x-forwarded-for` sometimes has multiple IP addresses because
+    # of redirects within our servers.
+    # The first IP is what we want, so remove everything after ",".
+    if (req.method == "PURGE" || req.method == "PURGE_REGEXP") {
+        if (!std.ip(regsub(req.http.x-forwarded-for, ",.*$", ""), "0.0.0.0") ~ purge) {
+            return (synth(405, "Not allowed."));
+        }
+        elsif (req.method == "PURGE") {
+          ban("req.url ~ " + req.url + "$");
+          return (synth(200, "Ban added"));
+        }
+        elsif (req.method == "PURGE_REGEXP") {
+          ban("req.url ~ " + req.url);
+          return (synth(200, "Regexp ban added"));
+        }
+    }
+
+    # Routing
     if (req.url ~ "^/ping") {
         set req.backend_hint = webview;
-        return (pass);
     }
-
-{% if 'press' in groups and groups['press'] %}
-    if (req.url ~ "^/api/") {
+    elsif (req.url ~ "^/api/") {
        set req.backend_hint = press_cluster.backend(req.http.cookie);
        # let the client talk directly to Press,
        # because litezip publishing payloads are huge.
        return (pipe);
     }
-{% endif %}
-
 {% if accounts_stub|default(False) %}
     # cnx rewrite stub login form
-    if (req.url ~ "^/stub-login-form") {
+    elsif (req.url ~ "^/stub-login-form") {
         set req.backend_hint = publishing_cluster.backend(req.http.cookie);
-        return (pass);
     }
-
 {% endif %}
     # cnx rewrite archive
-    if (req.url ~ "^/a/") {
+    elsif (req.url ~ "^/a/") {
         set req.backend_hint = publishing_cluster.backend(req.http.cookie);
-        return (pass);
     }
-
-    # cnx rewrite archive  - specials served from nginx statically
-    if (req.http.host ~ "^{{ arclishing_domain }}" || req.url ~ "^/sitemap.*.xml") {
-
+    elsif (req.url ~ "^/resources/") {
+        # Note: Medium-sized static files served directly from waitress
+        # We could pipe, but varnish is probably
+        # more suitable for serving these files than waitress
+        set req.backend_hint = resource_cluster.backend();
+    }
+    elsif ( req.url ~ "^/exports/") {
+        # Note: Large static files served directly from waitress
```

Comment:
Caching large files blows out the RAM cache, eviciting other small and medium files which are more likely to be served again. We should redirect these and serve from nginx, if possible.  The problem there is that archive has code to look up multiple names, and create a filesystem link to the appropriate other named file (legacy colXXX filenames linked as <UUID> names) We'd need logic to attempt to find via nginx, and if getting a 404, _then_ ask archive. Probably doable w/ some sort of restart. Need to break out the VCL state diagram. Not really a blocker for this PR, since this does seem to describe current behavior (I swear we used to do the other - I think inside nginx itself, somehow …)

Reply:
Yes, as you mentioned, I think this is the same as the current behavior.

Reply:
yeah, yeah, I guess. Just wanted to write it out. :-)